thejcpalma/palworld-dedicated-server-docker
# Add Custom RCON Broadcast Source Code

# Add Custom RCON Broadcast Source Code

## Summary

This pull request adds the custom RCON broadcast source code to the project, following the feature introduced in PR #5. The changes encompass the inclusion of essential code, dependencies, and updates to the Dockerfile to facilitate the build and integration of the custom RCON broadcast binary.

## Motivation and Context

The addition of the custom RCON broadcast feature, as outlined in PR #5, is needed for the capability to broadcast messages with spaces (not supported natively as spaces break the message).

After a lot of tries and using either underscores `_` or dashes `-` to concatenate the messages, I thought that it just isn't readable or pleasant at all.

I did some research with 'invisible' spaces (other characters that would replace the spaces but would be invisible) and ended up encountering this information about [ASCII Code for NBSP](https://www.ascii-code.com/character/nbsp) and a [proof of concept](https://github.com/Darkhand81/Palworld_broadcast_encoding_bug) by @Darkhand81 via [this Reddit post](https://www.reddit.com/r/Palworld/comments/1aplmvw/ive_figured_out_the_palworld_broadcast_command/) so I decided to code it a robust solution in Go so it could be compiled and used inside the container like @gorcon's rcon-cli.

## Description

The primary changes include:

- Introduction of the custom RCON broadcast source code to the project.
- Inclusion of necessary dependencies.
- Updates to the Dockerfile to ensure the build process incorporates the custom RCON broadcast binary.

## Testing Instructions

To validate these changes:

1. Build the project and ensure the custom RCON broadcast binary is successfully compiled.
2. Run the server with the new image.
3. Verify that the RCON broadcast command now supports spaces.

## Checklist
